### DISCUSSION TITLE
Question regarding detailed/formal documentation

### DISCUSSION THREAD

**Discussion by L015H4CK** [cmu-sei/GHOSTS#241] [Q&A] [answered]
Hello everyone,

I have recently found GHOSTS and was able to successfully set it up on a test machine. First off: Very important and great work!

When diving deeper into the configurations for the client (I am using the client without the API server - the possibility is mentioned [here](https://cmu-sei.github.io/GHOSTS/core/client/)) I stumbled upon several questions that I thought the documentation should answer.

For example: What are the possible "commands" for the various HandlerTypes? I found ["browse"](https://github.com/cmu-sei/GHOSTS/blob/master/src/Ghosts.Client/Sample%20Timelines/BrowserChrome.json), ["crawl" (only FF?)](https://github.com/cmu-sei/GHOSTS/blob/master/src/Ghosts.Client/Sample%20Timelines/Browser%20Crawl.json) and ["random"](https://cmu-sei.github.io/GHOSTS/core/client/#client-directory-structure) in various locations for "BrowserChrome", but did not find any documentation about them. Furthermore, what are valid "CommandArgs" for these different commands and how are they used? I found out that "random" visits random links on a visited website - where I initially thought it would randomly visit the urls defined in "CommandArgs" (Edit: this behavior depends on the "stickiness" arguments explained [here](https://github.com/cmu-sei/GHOSTS/discussions/119)). Again, I found some information and examples, but not a complete documentation.
(.. Any further questions are left out since this question should be about the documentation and not the questions themselves..)

Now to my question: Is there any "complete"/formal documentation for the timeline configuration that I missed (besides the documentation on [handlers](https://cmu-sei.github.io/GHOSTS/core/handlers/) and the examples) or is this still a "work in progress"?
At the moment it is quite tedious to test the different commands with different command args to see the results.

Thanks in advance!

Edit: Maybe all this is because I am not using the API Server and thus managing the client myself. Does the API Server show all the possible configurations for the clients?

**Comment by sei-dupdyke** [comment, marked as answer]
Hi, thanks for checking out the project. I updated [some of the documentation](https://cmu-sei.github.io/GHOSTS/core/handlers/browser/) in an attempt to answer some of your questions. Sorry about that.

**Reply by L015H4CK** [reply]
Thanks a lot for the quick response and added information! This is definitely helpful to me and hopefully to anyone that wants to check out the framework as well.
